lib/config.js

```diff
@@ -69,7 +72,53 @@ const getConfig = async (args) => {
     console.log('Error:', error.message);
   }
 
-  return mergeOptions(defaultOptions, contentfulCliOptions, environmentOptions, configFileOptions, args || {});
+  let userConfigFileOptions = {};
+  if (args.config) {
+    try {
+      let config = {};
+      let filepath = '';
+      if (args.config.endsWith('.js') || args.config.endsWith('.ts')) {
+        config = await jiti.import(path.resolve(args.config));
+        filepath = jiti.esmResolve(path.resolve(args.config));
+      } else {
+        const explorer = cosmiconfig('migrations');
+        const explorerResult = await explorer.load(args.config);
+        config = explorerResult.config;
+        filepath = explorerResult.filepath;
+      }
+      if (config) {
+        userConfigFileOptions = {
+          directory: path.resolve(path.dirname(filepath || ''), args.directory || 'migrations'),
+          ...(config || {}),
+        };
+
+        if (userConfigFileOptions.directory && !path.isAbsolute(userConfigFileOptions.directory)) {
+          userConfigFileOptions.directory = path.resolve(path.dirname(filepath || ''), userConfigFileOptions.directory);
+        }
+
+        if (configFileOptions.directory) {
+          delete userConfigFileOptions.directory;
+        }
```

Comment:
It's not clear why the config directory takes precedence over the user config directory. Would be nice to have a comment that explains that.

Reply:
you're right. this doesn't make sense